### ISSUE TITLE
swank:compile-file-if-needed throws 'OPERATION instances must only be created through MAKE-OPERATION'

### ISSUE THREAD

**Issue by d-m-racine** [slime/slime#415]
With latest SLIME on SBCL 1.4.2 / ASDF 3.3.1, swank:compile-file-if-needed with (pushnew 'try-compile-file-with-asdf *compile-file-for-emacs-hook*) in ~/.swank.lisp raises 'OPERATION instances must only be created through MAKE-OPERATION.'

This works as expected on the previous sbcl release (1.4.1 / ASDF 3.1.5) and may be a consequence of 'Enforce use of MAKE-OPERATION' in ASDF >= 3.1.7.32.
https://gitlab.common-lisp.net/asdf/asdf/commit/1e3471c607bf54c8b960071ac14534c4e7ad899b

Reproducible as follows with the system in the attached 'test.tar.gz':

[test.tar.gz](https://github.com/slime/slime/files/1540538/test.tar.gz)

; SLIME 2.20
CL-USER> (asdf:initialize-source-registry #P"~/test/asdf.conf")
; No value
CL-USER> (require :test)
; compiling file "/home/dracine/test/test.lisp" (written 06 DEC 2017 05:59:26 PM):
; compiling (IN-PACKAGE :CL-USER)
; compiling (FORMAT T ...)

; /home/dracine/.cache/common-lisp/sbcl-1.4.2-linux-x64/home/dracine/test/test-tmpGHU3ALSV.fasl written
; compilation finished in 0:00:00.001
*** Testing compile-and-load on SLIME 2.20 + SBCL 1.4.2 + ASDF 3.3.1 ***
NIL
CL-USER> ;; touch test.lisp or not, the following throws the same error
; No value
CL-USER> (swank:compile-file-if-needed "/home/dracine/test/test.lisp" t)

===
OPERATION instances must only be created through MAKE-OPERATION.
   [Condition of type ASDF/SESSION:FORMATTED-SYSTEM-DEFINITION-ERROR]

Restarts:
 0: [ABORT] Abort compilation.
 1: [RETRY] Retry SLIME REPL evaluation request.
 2: [*ABORT] Return to SLIME's top level.
 3: [ABORT] abort thread (#<THREAD "repl-thread" RUNNING {10025DFFA3}>)

Backtrace:
  0: (ASDF/SESSION:SYSDEF-ERROR "OPERATION instances must only be created through MAKE-OPERATION.")
  1: ((:METHOD INITIALIZE-INSTANCE :BEFORE (ASDF/OPERATION:OPERATION)) #<ASDF/LISP-ACTION:COMPILE-OP >) [fast-method]
  2: ((LAMBDA ()))
  3: (SWANK::TRY-COMPILE-FILE-WITH-ASDF #P"/home/dracine/test/test.lisp" T)
  4: ((LAMBDA NIL :IN SWANK:COMPILE-FILE-FOR-EMACS))
  5: ((LAMBDA NIL :IN SWANK::COLLECT-NOTES))
  6: (SWANK::MEASURE-TIME-INTERVAL #<CLOSURE (LAMBDA NIL :IN SWANK::COLLECT-NOTES) {1004B2FB5B}>)
  7: (SWANK::COLLECT-NOTES #<CLOSURE (LAMBDA NIL :IN SWANK:COMPILE-FILE-FOR-EMACS) {1004B2FB2B}>)
  8: (SWANK::CALL-WITH-BUFFER-SYNTAX NIL #<CLOSURE (LAMBDA NIL :IN SWANK:COMPILE-FILE-FOR-EMACS) {1004B2FAFB}>)
  9: (SB-INT:SIMPLE-EVAL-IN-LEXENV (SWANK:COMPILE-FILE-IF-NEEDED "/home/dracine/test/test.lisp" T) #<NULL-LEXENV>)
 10: (EVAL (SWANK:COMPILE-FILE-IF-NEEDED "/home/dracine/test/test.lisp" T))
 11: (SWANK::EVAL-REGION "(swank:compile-file-if-needed \"/home/dracine/test/test.lisp\" t) ..)
 12: ((LAMBDA NIL :IN SWANK-REPL::REPL-EVAL))
 13: (SWANK-REPL::TRACK-PACKAGE #<CLOSURE (LAMBDA NIL :IN SWANK-REPL::REPL-EVAL) {1004B2F34B}>)
 14: (SWANK::CALL-WITH-RETRY-RESTART "Retry SLIME REPL evaluation request." #<CLOSURE (LAMBDA NIL :IN SWANK-REPL::REPL-EVAL) {1004B2F2EB}>)
 15: (SWANK::CALL-WITH-BUFFER-SYNTAX NIL #<CLOSURE (LAMBDA NIL :IN SWANK-REPL::REPL-EVAL) {1004B2F2CB}>)
 16: (SWANK-REPL::REPL-EVAL "(swank:compile-file-if-needed \"/home/dracine/test/test.lisp\" t) ..)
 17: (SB-INT:SIMPLE-EVAL-IN-LEXENV (SWANK-REPL:LISTENER-EVAL "(swank:compile-file-if-needed \"/home/dracine/test/test.lisp\" t) ..)
 18: (EVAL (SWANK-REPL:LISTENER-EVAL "(swank:compile-file-if-needed \"/home/dracine/test/test.lisp\" t) ..)
 19: (SWANK:EVAL-FOR-EMACS (SWANK-REPL:LISTENER-EVAL "(swank:compile-file-if-needed \"/home/dracine/test/test.lisp\" t) ..)
 20: (SWANK::PROCESS-REQUESTS NIL)
 21: ((LAMBDA NIL :IN SWANK::HANDLE-REQUESTS))
 22: ((LAMBDA NIL :IN SWANK::HANDLE-REQUESTS))
 23: (SWANK/SBCL::CALL-WITH-BREAK-HOOK #<FUNCTION SWANK:SWANK-DEBUGGER-HOOK> #<CLOSURE (LAMBDA NIL :IN SWANK::HANDLE-REQUESTS) {100267805B}>)
 24: ((FLET SWANK/BACKEND:CALL-WITH-DEBUGGER-HOOK :IN "/home/dracine/common-lisp/extlib/slime-2.20/swank/sbcl.lisp") #<FUNCTION SWANK:SWANK-DEBUGGER-HOOK> #<CLOSURE (LAMBDA NIL :IN SWANK::HANDLE-REQUESTS) ..
 25: (SWANK::CALL-WITH-BINDINGS ((*STANDARD-INPUT* . #1=#<SWANK/GRAY::SLIME-INPUT-STREAM {10024E67F3}>) (*STANDARD-OUTPUT* . #2=#<SWANK/GRAY::SLIME-OUTPUT-STREAM {10025BFFA3}>) (*TRACE-OUTPUT* . #2#) (*ERR..
 26: (SWANK::HANDLE-REQUESTS #<SWANK::MULTITHREADED-CONNECTION {10035A22B3}> NIL)
 27: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
 28: ((FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE))
 29: ((FLET "WITHOUT-INTERRUPTS-BODY-1" :IN SB-THREAD::CALL-WITH-MUTEX))
 30: (SB-THREAD::CALL-WITH-MUTEX #<CLOSURE (FLET SB-THREAD::WITH-MUTEX-THUNK :IN SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE) {7FFFF4E16D2B}> #<SB-THREAD:MUTEX "thread result lock" owner: #<SB-THREAD:THR..
 31: (SB-THREAD::INITIAL-THREAD-FUNCTION-TRAMPOLINE #<SB-THREAD:THREAD "repl-thread" RUNNING {10025DFFA3}> NIL #<CLOSURE (LAMBDA NIL :IN SWANK-REPL::SPAWN-REPL-THREAD) {10025DFF4B}> (#<SB-THREAD:THREAD "re..
 32: ("foreign function: call_into_lisp")
 33: ("foreign function: new_thread_trampoline")

**Comment by attila-lendvai** [comment]
i cannot reproduce it with a fairly recent slime, although not head.

```
$ touch /tmp/test.lips
```

```
CL-USER> (swank:compile-file-if-needed "/tmp/test.lisp" t)
; compiling file "/tmp/test.lisp" (written 08 DEC 2017 12:04:07 AM):

; /tmp/test.fasl written
; compilation finished in 0:00:00.000
(:COMPILATION-RESULT NIL T 0.0 T "/tmp/test.fasl")
CL-USER> (swank:compile-file-if-needed "/tmp/test.lisp" t)
(:COMPILATION-RESULT NIL T 0.0 NIL NIL)
CL-USER> 
```

**Comment by luismbo** [comment]
@attila-lendvai do you have that `~/.swank.lisp` custom setting mentioned in the first paragraph?

**Comment by attila-lendvai** [comment]
no (i cannot parse that paragraph).

**Comment by luismbo** [comment]
I missed it too. Luckily, but the tar.gz has a more explicit mention in its README. Thanks for the useful recipe @d-m-racine, should be fixed in fd57c12.

**Comment by d-m-racine** [comment]
Thanks for the fix!  Sorry about that first sentence, it's quite an abomination.
